### PR TITLE
Reimplemented server response-waiting functionalities inside the methods of move_group_interface

### PR DESCRIPTION
**Pull request by CihatAltiparmak**
### Description

Firstly, this pull request is created in order to solve this issue [#2859](tg://search_hashtag?hashtag=2859) . I still have to apply some test such that this PR works for move2_tutorials and so on. I already have tested this PR on rviz but i need to try this part more to see if any corruption occurs for this PR.

I haven't tested computeCartesianMethod, getPlannerParams, setPlannerParams(acutally there is a posibility that i tested getPlannerParam and setPlannerParam implicitly while testing rviz and getInterfaceDescriptions, or else rviz gets stuck related to node spining)

I have removed callback_thread_ because it's not necessary to yield CPU continuously during move_group_interface to work. Instead of this, i just made a implementation for spinning only when client send a request from move_group_interface. I also have added private node for move_group_interface to handle its client jobs.

I also have to give anonymous node name to pnode_ . I will work on this as well.

I need some review from variable naming to logic. 

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/moveit/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/moveit/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by CihatAltiparmak** [comment]
Hello, it seems ready, but not ready for merging, i need a second eye. I only spin now when the methods like plan, execute etc. are called instead of spinning during lifetime of MoveGroupInterface instance. But I want to ask a question.

* With my modification, This instance(move_group_interface) cannot be used concurrently like this because of both of two threads will spin separately. In old implementation, it seems below pseudo code works. Should we also support to be able to implement codes like below?
```
move_group_interface = MoveGroupInterface(...)
thread1[move_group_interface] = {move_group_interface.plan()}
thread2[move_group_interface] = {move_group_interface.plan()}

thread1.start()
thread2.start()
```

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp [hunk 1]
```diff
@@ -833,13 +876,24 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     goal.controller_names = controllers;
 
     auto goal_handle_future = execute_action_client_->async_send_goal(goal, send_goal_opts);
+    if (callback_executor_.spin_until_future_complete(goal_handle_future,
+                                                      server_timeout.to_chrono<std::chrono::duration<double>>()) !=
+        rclcpp::FutureReturnCode::SUCCESS)
+    {
+      RCLCPP_ERROR(logger_, "MoveGroupInterface::execute() get sent call failed :(");
```
Comment (sjahr) [inline moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp:883]: You should make this error message more descriptive. Try to explain what failed and why

Comment (sjahr) [reply moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp:883]: You'll probably need to update all error messages like this

**Review by sjahr — COMMENTED** [review]
Thanks! How have you tested this? With the move_group tutorial?

**Pull request closed without merging**